### DISCUSSION TITLE
Custom source indexes should have a checkbox column to provide basic actions

### DISCUSSION THREAD

**Discussion by kylecotter** [craftcms/cms#11383] [Ideas] [answered]
Similar to "All Entries", would be nice if custom sources had a checkbox column and an actions menu for toggling the status, selecting multiple entries, and edit, view, delete, etc.

**Comment by brandonkelly** [comment, marked as answer]
Just made this happen for Craft 4.1 🎉  (560bf35a030605838d001673f7acb6d4b2cc4afc)

Going forward, all element sources will get the following actions added automatically:

- Set Status (for element types with statuses)
- View (for element types with URIs)
- Edit
- Delete

Set Status, Edit, and Delete will only actually be usable on elements depending on their `canSave()` and `canDelete()` methods.

**Reply by brandonkelly** [reply]
Craft 4.1 is out with this change ✨
